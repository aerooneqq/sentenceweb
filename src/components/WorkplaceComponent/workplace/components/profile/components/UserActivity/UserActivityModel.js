import React from "react"
import MonthActivityComponent from "./userActivityAtoms/MonthActivity/MonthActivity";
import DateService from "../../../../../../../services/Dates/DateService";

export default class UserActivityModel { 
    months = [
        {name: "January", daysCount: 31, firstDayNum: 1}, 
        {name : "February", daysCount: 28, firstDayNum: 4}, 
        {name : "March", daysCount: 31, firstDayNum: 4}, 
        {name: "April", daysCount: 30, firstDayNum: 0}, 
        {name : "May", daysCount: 31, firstDayNum: 2}, 
        {name : "June", daysCount: 30, firstDayNum: 5}, 
        {name : "July", daysCount: 31, firstDayNum: 0},
        {name : "August", daysCount: 31, firstDayNum: 3}, 
        {name :"September", daysCount: 30, firstDayNum: 6}, 
        {name : "October", daysCount: 31, firstDayNum: 1}, 
        {name : "November", daysCount: 30, firstDayNum: 4},
        {name : "December", daysCount: 31, firstDayNum: 6}
    ];

    constructor(activities) { 
        this._activities = activities;
    }

    getActitivtyComponents() { 
        let data = this._activities;

        let monthsActivities = this._createEmptyCalendar();

        for (let activity of data.activities) { 
            let dateService = new DateService(activity.activityDate);
            
            let monthIndex = dateService.getMonth() - 1;
            let dayIndex = dateService.getDay() - 1;

            monthsActivities[monthIndex][dayIndex].count += 1;
            monthsActivities[monthIndex][dayIndex].activities.push(activity);
        }

        let monthActivitiesComponents = [];

        for (let index in this.months) { 
            monthActivitiesComponents.push(
                <MonthActivityComponent daysCount = {this.months[index].daysCount}
                                        activitiesData = {monthsActivities[index]}
                                        month = {this.months[index]} />
                )
        }

        return monthActivitiesComponents;
    }

    _createEmptyCalendar() { 
        let monthActivities = []

        for (let i = 0; i < 12; i++) { 
            monthActivities.push({});

            for (let j = 0; j < this.months[i].daysCount; j++) {
                monthActivities[i][j] = {count: 0, activities: []};
            }
        }

        return monthActivities;
    }

}