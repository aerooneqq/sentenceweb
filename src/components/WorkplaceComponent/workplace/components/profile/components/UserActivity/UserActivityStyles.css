@media screen and (min-width: 400px){ 
    #userActivityOutterContainer{ 
        flex-grow: 1;
    
        display: flex;
        flex-direction: column;
    }
    
    #userActivitiesCont{
        height: 200px;
        overflow-y: hidden;
        overflow-x: scroll;
        
        display: flex;
    
        box-sizing: content-box;
    
        background: #f9f9f9;
    
        border-style: none; 
    }
    
    #userActivitiesCont::-webkit-scrollbar{ 
        height: 5px;
    }   
    
    #userActivitiesCont::-webkit-scrollbar-thumb{
        background: #F97646;
        border-radius: 2px;
    }
    
    .monthActivityCont:nth-child(n){ 
        margin-right: 5px;
    }
    
    #activityHeader{ 
        font-family: Arial; 
        color: #F97646;
        font-size: 20px;
    }
}

@media screen and (max-width: 400px){ 
    #userActivityOutterContainer{ 
        display: none;
    }
}