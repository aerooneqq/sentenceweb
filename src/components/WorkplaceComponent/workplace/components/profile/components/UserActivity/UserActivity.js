import React, {Component, lazy, Suspense} from "react";
import axios from "axios";

//Styles
import "./UserActivityStyles.css";

//Services
import UserActivitiesService from "../../../../../../../services/UserServices/UserActivitiesService";
import DateService from "../../../../../../../services/Dates/DateService";

//App messages
import {alertAppMessage} from "../../../../../../ApplicationMessage/ApplicationMessageManager";

//Components
const MonthActivityComponent = lazy(() => import("./userActivityAtoms/MonthActivity/MonthActivity"));
const Loader = lazy(()=>import("../../../../../../loader/Loader"));
const ProfileHeader = lazy(() => import(("../ProfileHeader/ProfileHeader")));

export default class UserActivity extends Component{
    months = [
        {name: "January", daysCount: 31, firstDayNum: 1}, 
        {name : "February", daysCount: 28, firstDayNum: 4}, 
        {name : "March", daysCount: 31, firstDayNum: 4}, 
        {name: "April", daysCount: 30, firstDayNum: 0}, 
        {name : "May", daysCount: 31, firstDayNum: 2}, 
        {name : "June", daysCount: 30, firstDayNum: 5}, 
        {name : "July", daysCount: 31, firstDayNum: 0},
        {name : "August", daysCount: 31, firstDayNum: 3}, 
        {name :"September", daysCount: 30, firstDayNum: 6}, 
        {name : "October", daysCount: 31, firstDayNum: 1}, 
        {name : "November", daysCount: 30, firstDayNum: 4},
        {name : "December", daysCount: 31, firstDayNum: 6}];
    
    constructor(props){ 
        super(props)

        this.state = { 
            components: <Loader message = "Loading activities..." />,
            width: 500
        };
    }

    componentDidMount(){ 
        this.setState({ 
            components: <Loader message = "Loading activities" />
        })

        let userActivitiesService = new UserActivitiesService();
        userActivitiesService.getUserActivities(localStorage.getItem("token"))
            .then(res => 
            { 
                let data = res.data;
                let monthActivities = data.activities.map(activity => activity.activityDate);
                
                let yearActivities = {}

                for (let i = 0; i<12; i++){ 
                    yearActivities[i] = {}
                }

                for (let index in monthActivities) { 
                    let dateService = new DateService(monthActivities[index]);

                    let day = dateService.getDay() - 1;
                    let month = dateService.getMonth() - 1;

                    if (yearActivities[month][day] === undefined){
                        yearActivities[month][day] = 1;
                    }
                    else { 
                        yearActivities[month][day] += 1;
                    }
                }

                let yearActivitiesComponentList = []
                for (let month in this.months){ 
                    yearActivitiesComponentList[month] = [];
                }

                let monthIndex = 0;
                for (let month in this.months){ 
                    for (let i = 0; i<31; i++){ 
                        yearActivitiesComponentList[month].push(
                            {count: yearActivities[monthIndex][i] === undefined ? 
                            0 : yearActivities[monthIndex][i]})
                    }
                    monthIndex+=1;
                }

                monthActivities = []

                for (var key in this.months){ 
                    monthActivities.push(
                        <div className = "monthActivityCont">
                            <MonthActivityComponent daysCount = {yearActivitiesComponentList[key].length}
                            activities = {yearActivitiesComponentList[key]} month = {this.months[key]}/>
                        </div>
                    )
                }

                this.setState({ 
                    components: monthActivities,
                    width: document.getElementById("userActivityOutterContainer").offsetWidth - 100
                });
            }).catch(er => { 
                alertAppMessage("Error occured while getting your activities", "error");
            });
    }

    render(){ 
        return(
            <div id = "userActivityOutterContainer">
                <div id = "userActivitiesCont" 
                     className = "profileShadowContainer">
                    <Suspense fallback = {<Loader message = "Loading activities..." />}>
                        {this.state.components}
                    </Suspense>
                </div>
            </div>
        )
    }
}