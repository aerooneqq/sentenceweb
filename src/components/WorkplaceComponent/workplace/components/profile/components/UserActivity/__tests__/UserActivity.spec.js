import React from "react";
import { shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import UserActivity from "../UserActivity";

describe("UserActivity", () => { 
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<UserActivity />))).toMatchSnapshot();
    });

    it("should have proper initial state", () => { 
        const properState = { 
            isFirstLoad: true
        }

        const userActivity = shallow(<UserActivity />);

        for (let key in properState) { 
            expect(properState[key]).toBe(userActivity.state(key));
        }
    });
});