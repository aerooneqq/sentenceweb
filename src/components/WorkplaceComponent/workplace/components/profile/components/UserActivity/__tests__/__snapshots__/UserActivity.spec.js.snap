// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserActivity should render correctly 1`] = `
<div
  id="userActivityOutterContainer"
>
  <div
    className="profileShadowContainer"
    id="userActivitiesCont"
  >
    <Suspense
      fallback={
        <UNDEFINED
          message="Loading activities..."
        />
      }
    >
      <UserActivityLoader />
    </Suspense>
  </div>
</div>
`;
